### ISSUE TITLE
Add support for Lambda Destinations for Asynchronous Invocations

### ISSUE THREAD

**Issue by sammefford** [terraform-providers/terraform-provider-aws#11087]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Lambdas will be significantly faster (less expensive) and simpler (no more routing or retries in lambda code) with the addition of [Destinations for Asynchronous Invocations](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-supports-destinations-for-asynchronous-invocations/).

### New or Affected Resource(s)

* aws_lambda_function
* aws_lambda_event_destination_mapping (new)

### Potential Terraform Configuration

```hcl
resource "aws_lambda_function" "this" {
    ...
    event_invoke_config {
        maximum_retry_attempts = "..."
        maximum_event_age_in_seconds = "..."
    }
}

resource "aws_lambda_destination" "this" {
    function_name = "..."
    destination_config = {
        on_success = {
            destination = "..."
        }
        on_failure = {
            destination = "..."
        }
    }
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-supports-destinations-for-asynchronous-invocations/
* https://aws.amazon.com/blogs/compute/introducing-aws-lambda-destinations/
* https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html
* https://docs.aws.amazon.com/lambda/latest/dg/API_PutFunctionEventInvokeConfig.html
* https://docs.aws.amazon.com/cli/latest/reference/lambda/put-function-event-invoke-config.html

**Comment by bflad** [comment]
I'll pick this one up since it is similar in configuration and testing to the new [Lambda Provisioned Concurrency configuration](https://github.com/terraform-providers/terraform-provider-aws/issues/11128). Since the API refers to these as "FunctionEventInvokeConfig", we'll probably stick with an admittedly uglier `aws_lambda_function_event_invoke_config` resource naming. We'll also want to introduce this solely as a separate resource to prevent multiple Terraform resources from being able to manage the same underlying infrastructure configuration.

**Comment by bflad** [comment]
Pull request submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/11165

**Comment by bflad** [comment]
Support for the new `aws_lambda_function_event_invoke_config` resource has been merged and will release with version 2.43.0 of the Terraform AWS Provider, Thursday this week. 👍
